test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/ObjDoubleConsumerTest.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.{Arrays, ArrayList, List => JavaList}
+import java.util.function.{Consumer, ObjDoubleConsumer}
+import org.junit.Assert._
+import org.junit.Test
+
+class ObjDoubleConsumerTest {
+  import ObjDoubleConsumerTest._
+
+  @Test def accept(): Unit = {
+    val arr: JavaList[Double] = Arrays.asList(3.0d, 2.0d, 5.0d, 7.0d, 4.0d)
+
+    // Side-effects
+    var current: ArrayList[Double] = new ArrayList[Double](arr.size())
+    val double: ObjDoubleConsumer[JavaList[Double]] = makeConsumer { (t: JavaList[Double], num) =>
+      t.forEach(
+        new Consumer[Double] {
+          def accept(value: Double): Unit = {
+            current.add(value * num)
+            ()
+          }
+        }
+      )
+    }
```

Comment:
That is a fairly complicated test, which involves a number of moving parts that are unrelated to `ObjDoubleConsumer`: `java.util.List`, its `forEach` method, the generic `Consumer[T]`.

Consider a test where the `ObjDoubleConsumer` simply stores its two arguments in an enclosing `var result: (SomeClass, Double)`. There's no need to involve the collections here.

(of course the same comment applies to the other `ObjXConsumerTest`s.